common/src/main/java/com/redhat/cloud/notifications/routers/sources/SecretUtils.java

```diff
@@ -300,4 +345,18 @@ protected boolean isBasicAuthNullOrBlank(final BasicAuthentication basicAuthenti
         return (basicAuthentication.getPassword() == null || basicAuthentication.getPassword().isBlank()) &&
                 (basicAuthentication.getUsername() == null || basicAuthentication.getUsername().isBlank());
     }
+
+    /**
+     * Prepends the endpoint's UUID to the thrown {@link WebApplicationException}
+     * in the {@link SourcesService} to make it easier to debug what the
+     * problem might have been. Then, it rethrows the exception.
+     * @param endpointUuid the endpoint's UUID to prepend.
+     * @param e the exception that was thrown by the {@link SourcesService}.
+     */
+    protected void prependEndpointUuidToExceptionAndRethrow(final UUID endpointUuid, final WebApplicationException e) {
+        throw new WebApplicationException(
+            String.format("[endpoint_uuid: %s]%s", endpointUuid, e.getMessage()),
+            e.getResponse()
+        );
+    }
```

Comment:
I was thinking that maybe an alternative could be the following one:

```
final UUID debugUuid = UUID.random();

Log.errorf(String.format("[debug_uuid: %s][endpoint_uuid: %s]%s", debugUuid, endpointUuid, e.getMessage());

throw new WebApplicationException(String.format("an error occurred when reaching out Sources. Debug UUID: %s", debugUuid));
```
Since the consumers are not interested in this much detail? Although I don't know if it is the best moment to get exquisite about this...